internal/appliance/reconciler/frontend_test.go

```diff
@@ -16,3 +26,54 @@ func (suite *ApplianceTestSuite) TestDeployFrontend() {
 		})
 	}
 }
+
+func (suite *ApplianceTestSuite) TestAdoptFrontend() {
+	namespace, err := k8senvtest.NewRandomNamespace("test-appliance")
+	suite.Require().NoError(err)
+	_, err = suite.k8sClient.CoreV1().Namespaces().Create(suite.ctx, namespace, metav1.CreateOptions{})
+	suite.Require().NoError(err)
+	testService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sourcegraph-frontend",
+			Namespace: namespace.Name,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports:    []corev1.ServicePort{{Name: "http", Port: 30080, TargetPort: intstr.FromString("http")}},
+			Selector: map[string]string{"app": "sourcegraph-appliance"},
+		},
+	}
+	_, err = suite.k8sClient.CoreV1().Services(namespace.Name).Create(suite.ctx, &testService, metav1.CreateOptions{})
+	suite.Require().NoError(err)
+
+	testIngress := ingress.NewIngress("sourcegraph-frontend", namespace.Name)
+	testIngress.Spec.Rules = []netv1.IngressRule{{
+		Host: "an-existing-hostname.com",
+		IngressRuleValue: netv1.IngressRuleValue{
+			HTTP: &netv1.HTTPIngressRuleValue{
+				Paths: []netv1.HTTPIngressPath{{
+					Path:     "/",
+					PathType: pointers.Ptr(netv1.PathTypePrefix),
+					Backend: netv1.IngressBackend{
+						Service: &netv1.IngressServiceBackend{
+							Name: "sourcegraph-frontend",
+							Port: netv1.ServiceBackendPort{
+								Number: 30081,
+							},
+						},
+					},
+				}},
+			},
+		},
+	}}
+	ingressClassName := "nginx"
+	testIngress.Spec.IngressClassName = &ingressClassName
+	_, err = suite.k8sClient.NetworkingV1().Ingresses(namespace.Name).Create(suite.ctx, &testIngress, metav1.CreateOptions{})
+	suite.Require().NoError(err)
+
+	cfgMap := suite.newConfigMap(namespace.GetName(), "frontend/adopt-service")
```

Comment:
Optional nit: just use the default input fixture, and delete sg/frontend/adopt-service.yaml (since these files are identical) - the adoption does occur by default. Would still need the special-case golden (output) fixture of course, so line 78 wouldn't change.

```suggestion
	cfgMap := suite.newConfigMap(namespace.GetName(), "frontend/default")
```

Reply:
Switched to using `with-ingress` since we're using an ingress, instead of the default one